pkg/agent/controller/servicecidr/provider.go

```diff
@@ -0,0 +1,150 @@
+// Copyright 2023 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package servicecidr
+
+import (
+	"fmt"
+	"net"
+	"sync"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	coreinformers "k8s.io/client-go/informers/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog/v2"
+	utilnet "k8s.io/utils/net"
+
+	"antrea.io/antrea/pkg/agent/util"
+)
+
+const (
+	// Disable resyncing.
+	resyncPeriod time.Duration = 0
+)
+
+type EventHandler func(serviceCIDRs []*net.IPNet)
+
+type Interface interface {
+	GetServiceCIDR(isIPv6 bool) (*net.IPNet, error)
+	// The added handlers will be called when Service CIDR changes.
+	AddEventHandler(handler EventHandler)
+}
+
+type provider struct {
```

Comment:
How about "discoverer" or "detector"?

Reply:
Sure, this part is from #4570 @hongliangl could you help to rename it? thanks.

Reply:
I'll rename that.